src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/DateTimeFieldDriver.cs

```diff
@@ -0,0 +1,43 @@
+using System.Threading.Tasks;
+using OrchardCore.ContentFields.ViewModels;
+using OrchardCore.ContentManagement.Display.ContentDisplay;
+using OrchardCore.ContentManagement.Display.Models;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Views;
+
+namespace OrchardCore.ContentFields.Fields
+{
+    public class DateTimeFieldDisplayDriver : ContentFieldDisplayDriver<DateTimeField>
+    {
+        public override IDisplayResult Display(DateTimeField field, BuildFieldDisplayContext context)
+        {
+            return Initialize<DisplayDateTimeFieldViewModel>("DateTimeField", model =>
+            {
+                model.Field = field;
+                model.Part = context.ContentPart;
+                model.PartFieldDefinition = context.PartFieldDefinition;
+            })
+            .Location("Content")
+            .Location("SummaryAdmin", "");
+            ;
+        }
+
+        public override IDisplayResult Edit(DateTimeField field, BuildFieldEditorContext context)
+        {
+            return Initialize<EditDateTimeFieldViewModel>("DateTimeField_Edit", model =>
+            {
+                model.Value = field.Value;
+                model.Field = field;
+                model.Part = context.ContentPart;
+                model.PartFieldDefinition = context.PartFieldDefinition;
+            });
+        }
+
+        public override async Task<IDisplayResult> UpdateAsync(DateTimeField field, IUpdateModel updater, UpdateFieldEditorContext context)
+        {
+            await updater.TryUpdateModelAsync(field, Prefix, f => f.Value);
```

Comment:
Create a new `EditDateTimeFieldViewModel` instance and bind the `Value` property.
Then convert it to UTC (from the current time zone) and assign to `field.Value`.

Reply:
`field.Value = new DateTimeOffset(model.Value,  _localClock.GetTimeZoneAsync()).ToUniversal().DateTime;`